packages/flutter/lib/src/cupertino/button.dart

```diff
@@ -9,21 +9,85 @@ import 'package:flutter/foundation.dart';
 import 'package:flutter/widgets.dart';
 
 import 'colors.dart';
-import 'constants.dart';
+import 'text_theme.dart';
 import 'theme.dart';
 
-// Measured against iOS 12 in Xcode.
-const EdgeInsets _kButtonPadding = EdgeInsets.all(16.0);
-const EdgeInsets _kBackgroundButtonPadding = EdgeInsets.symmetric(
-  vertical: 14.0,
-  horizontal: 64.0,
-);
 
-// The relative values needed to transform a color to it's equivalent focus
-// outline color.
-const double _kCupertinoFocusColorOpacity = 0.80;
-const double _kCupertinoFocusColorBrightness = 0.69;
-const double _kCupertinoFocusColorSaturation = 0.835;
+// Measured against iOS 17 guidelines (https://developer.apple.com/design/human-interface-guidelines/buttons#iOS-iPadOS)
+
+/// The size of a [CupertinoButton].
+/// Based on the iOS (17) Human Interface Guidelines (https://developer.apple.com/design/human-interface-guidelines/buttons#iOS-iPadOS)
+enum CupertinoButtonSize {
+  /// Displays a smaller button with round sides and smaller text (uses [CupertinoTextThemeData.actionSmallTextStyle]).
+  small,
+  /// Displays a medium sized button with round sides and regular-sized text
```

Comment:
Nit that will inevitably come up at some point: all comments should end in a period.